universal-application-tool-0.0.1/test/controllers/admin/QuestionControllerTest.java

```diff
@@ -183,18 +185,22 @@ public void newOne_returnsFailureForInvalidQuestionType() {
 
   @Test
   public void update_redirectsOnSuccess() {
-    Question question = resourceCreator().insertQuestion("my.path.name");
+    QuestionDefinition nameQuestion = TestQuestionBank.applicantName().getQuestionDefinition();
     ImmutableMap.Builder<String, String> formData = ImmutableMap.builder();
     formData
-        .put("questionName", "name")
-        .put("questionDescription", "desc")
-        .put("questionParentPath", "my.path")
-        .put("questionType", "TEXT")
-        .put("questionText", "question text updated!")
-        .put("questionHelpText", ":-)");
+        .put("questionName", nameQuestion.getName())
+        .put("questionDescription", "a new description")
+        .put("questionParentPath", nameQuestion.getPath().parentPath().toString())
+        .put("questionType", nameQuestion.getQuestionType().name())
+        .put("questionText", "question text updated")
+        .put("questionHelpText", "a new help text");
     RequestBuilder requestBuilder = addCSRFToken(Helpers.fakeRequest().bodyForm(formData.build()));
 
-    Result result = controller.update(requestBuilder.build(), question.id, "text");
+    Result result =
+        controller.update(
+            requestBuilder.build(),
+            nameQuestion.getId(),
+            nameQuestion.getQuestionType().toString());
```

Comment:
Question (non-blocking): what's the difference between `#name` and `#toString`? Should we have only one?